util/fibers/synchronization.h

```diff
@@ -422,12 +427,19 @@ class BlockingCounter {
         ec_.notifyAll();
     }
 
+    std::optional<bool> IsReady() const {
+      auto cnt = count_.load(memory_order_acquire);
+      return (cnt == 0 || (cnt & kCancelFlag)) ? std::make_optional((cnt & kCancelFlag) == 0)
+                                               : nullopt;
+    }
```

Comment:
Because we don't use the event-count here anymore (which provides acquire-release semantics), we actually have to use acquire here... But others _could_ still be relaxed because notify provides a release fence 🤔